stackrox/acs-fleet-manager
# Support deploying Operator from Quay on remote OpenShift Cluster

## Description

Until now it was only possible to deploy non-marketplace (i.e. Quay) operator on local clusters (e.g. Colima).
With this change it is possible to also deploy such non-release operators to remote OpenShift clusters.

In case we would need to use a non-release operator in the future it is good to have this capability also during testing.

## Checklist (Definition of Done)

- [ ] ~~Documentation added if necessary (i.e. changes to dev setup, test execution, ...)~~ (n/a, since this fixes an undocumented limitation).

## Test manual

1. Spin up remote OpenShift cluster
2. Make sure `QUAY_USER` and `QUAY_TOKEN` is set in your environment.
3. Make sure that you have your `AUTH_TYPE` set up correctly (note: there is currently a bug regarding `STATIC_TOKEN`...).
4. Use for example `export FLEET_MANAGER_IMAGE=quay.io/mclasmei/fleet-manager:v0.0.0-357-gcb53739d8a` (when deploying to remote cluster we need to use fleet-manager images pullable from Quay.)
5. Exec `OPENSHIFT_MARKETPLACE=false OPERATOR_SOURCE=quay dev/env/scripts/bootstrap.sh`
6. Verify that the operator is running in the namespace `stackrox-operator`.
